otp4gb/parameters.py

```diff
@@ -0,0 +1,526 @@
+    ]
+
+    matched_zones = zones[np.isin(zones, lookup.index)]
+    unmatched_zones = zones[~ np.isin(zones, lookup.index)]
+
+    if (len(matched_zones) == len(zones)) & (len(unmatched_zones) == 0):
+        # All zones have been matched to a ruc code.
+        # Now, make ruc_lookup contain only zones ids that are within `zones` for when we
+        #   apply the distance weighting calculation
+
+        lookup_data = lookup_data.loc[lookup_data[data.id_column].isin(matched_zones)]
+        lookup: pd.Series = lookup_data.set_index(data.id_column, verify_integrity=True)[
+            data.ruc_column
+        ]
+
+        LOG.info(f"{len(zones)} zones matched to RUC classifications successfully")
+
+    else:
+        raise ValueError(
+            f"{len(zones)} centroids supplied but only {len(matched_zones)} matches "
+            f"with the supplied ruc lookup. {len(unmatched_zones)} zone ids are missing"
+            f"from the lookup. Add these... {_summarise_list(unmatched_zones)}"
+        )
+
+    # TODO(MB) Allow RUC lookup to include custom factors instead of RUC codes
+    lookup = lookup.astype(str).str.upper()
+    invalid_ruc = lookup[~lookup.isin(RUC_WEIGHTS)].unique().tolist()
+    if len(invalid_ruc) > 0:
+        raise ValueError(
+            f"{len(invalid_ruc)} values in RUC classifications not "
+            f"found in centroids data: {_summarise_list(invalid_ruc)}"
+        )
+
+    missing_zones = zones[~np.isin(zones, lookup.index)]
+    if len(missing_zones) > 0:
+        LOG.warning(
+            "RUC distance factor defaulting to 1 for %s zones: %s",
+            len(missing_zones),
+            _summarise_list(missing_zones),
+        )
+        lookup = pd.concat([lookup, pd.Series(data=1, index=missing_zones)])
+
+    lookup = lookup.replace(RUC_WEIGHTS).astype(float)
+
+    return lookup
+
+
+def _load_previous_trips(data: PreviousTrips) -> set:
+    """Load previously requested trips into a python set"""
+    # TODO(MB): remove the check_column from this if continuing to use a set of od_pairs
+    previous_trips = pd.read_csv(data.path, usecols=[data.od_column, data.check_column])
+
+    # Convert lookup to a set - think it will be faster to compare items to a set rather than a lookup
+    previous_od_trips = set(previous_trips[data.od_column])
+
+    return previous_od_trips
+
+
+def _load_irrelevant_destinations(
+        data: IrrelevantDestinations, zones: np.ndarray
+) -> np.ndarray | None:
+    """Load array of destinations to exclude, return None if file is empty."""
+    LOG.info("Loading irrelevant destinations from %s", data.path.name)
+    irrelevant_data = pd.read_csv(ASSET_DIR / data.path, usecols=[data.zone_column])
+
+    irrelevant = irrelevant_data[data.zone_column].unique()
+    matched_irrelevant = np.isin(irrelevant, zones)
+
+    LOG.info("Given %s unique destinations to exclude",
+             len(matched_irrelevant)
+             )
+
+    if len(irrelevant) == 0:
+        return None
```

Comment:
This should check `matched_irrelevant`

```suggestion
    if len(matched_irrelevant) == 0:
        return None
```

Reply:
[33618e0](https://github.com/Transport-for-the-North/otp4gb-py/commit/33618e04ef1241c97b278a6e909bd1692db126a5)

Changed the return None check to be on matched_irrelevant. However, kept the check on irrelevant, which now returns a ValueError if the length of the file is 0, prompting a user to provide a file with some data.

Reply:
That's a great idea